test/mock_ioctl_linux.c

```diff
@@ -6,11 +6,20 @@
 #include <stdio.h>
 #include <sys/ioctl.h>
 
+/* Work around differing ioctl signatures between alpine and debian */
+#ifdef __GLIBC__
 typedef int (*ioctl_t)(int, unsigned long, char*);
+#else
+typedef int (*ioctl_t)(int, int, char*);
```

Comment:
What about the following:

```c
#ifdef __GLIBC__
typedef unsigned long request_t;
#else
typedef int request_t;
#endif

typedef int (*ioctl_t)(int, request_t, char *);

int ioctl(int fd, request_t request, ...)
{
    ...
}
```

It would avoid using the preprocessor on entire lines, and would be needed once instead of twice.

Reply:
Nice improvement - thanks!